### PR TITLE
Db check duplicates

### LINKED ISSUE #574: Use Github IDs instead of names in DB
Currently user, repository, organization are all stored by their name in the database. User, repo or organization can change their name and I guess this is not that common but I still think it would be better to store IDs that we get from the Github API instead of names. 

---

- [ ] Store and use user ID. 
- [ ] Store and save username. 
- [ ] Ensure that user name is updated every time user ID is accessed, in case a username is updated.

### PR DESCRIPTION
**Pull request by kamaalsultan**
Resolves #574

### CODE REVIEW

**Review by 0x4007 — COMMENTED** [review]
I just updated the specification, but I don't see a mechanism to sync the username. Do you have that here?

**Comment by byteballet** [comment]
> I just updated the specification, but I don't see a mechanism to sync the username. Do you have that here?

Sorry, I thought it will be a new issue. I will update.

**Comment by pavlovcik** [comment]
> > I just updated the specification, but I don't see a mechanism to sync the username. Do you have that here?
> 
> 
> 
> Sorry, I thought it will be a new issue. I will update.

That might have made sense actually. Hopefully it's fine I updated the existing one!

**Comment by pavlovcik** [comment]
@rndquu rfc

I figured caching the username for reads is useful to save on network requests. It seems useful to be able to directly refer to the username from the database.

**Comment by byteballet** [comment]
> I figured caching the username for reads is useful to save on unnecessary network requests to the GitHub API. Intuitively to me it seems useful to be able to directly refer to the username from the database. But at the same time, perhaps it's a waste of storage space.

Also I think there is no big difference between reading database and using github api.

**Comment by whilefoo** [comment]
I took a quick look and we never get username from database. Most of the time all user data is available in the webhook payload.

File: package-lock.json
Comment (pavlovcik) [inline package-lock.json:1]: This should not be here. We use yarn. Please remove.

File: src/handlers/comment/handlers/query.ts [hunk 1]
```diff
@@ -22,9 +23,11 @@ export const query = async (body: string) => {
   const regex = /^\/query\s+@([\w-]+)\s*$/;
   const matches = body.match(regex);
   const user = matches?.[1];
+  getUser;
```
Comment (pavlovcik) [inline src/handlers/comment/handlers/query.ts:26]:
```suggestion
```

**Comment by pavlovcik** [comment]
> I took a quick look and we never query username from database. Most of the time all user data is available in the webhook payload.

Looks like we should get rid of the username entirely then.

**Comment by byteballet** [comment]
> Looks like we should get rid of the username entirely then.

Yes, that must be the solution.

**Comment by Draeieg** [comment]
let me know when this is ready to go into QA

**Comment by byteballet** [comment]
> let me know when this is ready to go into QA

I guess it's now.

**Comment by rndquu** [comment]
@byteballet Could you resolve the conflicts one more time?

@whilefoo @wannacfuture Could you help with the review?

**Review by wannacfuture — COMMENTED** [review]
@byteballet  Could you provide QA? Just wanted to make sure if its working as normal.

File: src/handlers/comment/handlers/query.ts [hunk 2]
```diff
@@ -24,7 +25,8 @@ export const query = async (body: string) => {
   const user = matches?.[1];
```
Comment (whilefoo) [inline src/handlers/comment/handlers/query.ts:25]:
```suggestion
  const username = matches?.[1];
```

File: src/handlers/comment/handlers/query.ts [hunk 3]
```diff
@@ -24,7 +25,8 @@ export const query = async (body: string) => {
...
 
   if (user) {
-    const walletInfo = await getWalletInfo(user, id?.toString());
+    const { id: user_id } = (await getUser(user)) ?? {};
```
Comment (whilefoo) [inline src/handlers/comment/handlers/query.ts:28]:
```suggestion
    const user = await getUser(user);
    if (!user) {
      logger.info(`Error retrieving data about user @${username}`);
      return `Error retrieving data about user @${username}`;
    }
```

File: src/handlers/comment/handlers/query.ts [hunk 4]
```diff
@@ -24,7 +25,8 @@ export const query = async (body: string) => {
...
+    const walletInfo = await getWalletInfo(user, id?.toString(), user_id || 0);
```
Comment (whilefoo) [inline src/handlers/comment/handlers/query.ts:29]:
```suggestion
    const walletInfo = await getWalletInfo(user, id.toString(), user.id);
```

**Review by whilefoo — CHANGES_REQUESTED** [review]
Code looks mostly good. Please post QA.

**Review by cuddle-dev — CHANGES_REQUESTED** [review]

File: supabase/migrations/20230821160747_add_id.sql [hunk 5]
```diff
@@ -0,0 +1,27 @@
+ALTER TABLE wallets
```
[the remark this answers is not in the document]
Comment (byteballet) [reply supabase/migrations/20230821160747_add_id.sql:1]: Okay, I will have a look.
Thanks

**Pull request closed without merging**